core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -185,13 +185,14 @@ class Analyzer(
       sender ! VoidResponse
     case TypecheckModule(moduleName) =>
       //consider the case of a project with no modules
-      config.modules get (moduleName) foreach {
+      //hack: to typecheck both compile and test configurations
+      (config.modules filter { case (moduleId, module) => moduleId.project == moduleName }).values foreach {
         module =>
           val files: Set[SourceFileInfo] = module.scalaSourceFiles.map(s => SourceFileInfo(EnsimeFile(s), None, None))(breakOut)
           sender ! scalaCompiler.handleReloadFiles(files)
       }
     case UnloadModuleReq(moduleName) =>
-      config.modules get (moduleName) foreach {
+      (config.modules filter { case (moduleId, module) => moduleId.project == moduleName }).values foreach {
```

Comment:
You can use `filterKeys` to simplify the code a bit. `filterKeys(id => id.project == moduleName)`.